### PR TITLE
Fix TT2 button styles

### LINKED ISSUE #37578: Fix TT2 button styles and introduce missing button classes for the order-again and orders pagination buttons
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Currently, TT2 contains the following:
- Hardcoded colors in buttons styles
- Misalignment with hardcoded color variables on the CSS file vs. the theme.json file.

This issue aims to keep track of the following two fixes:
1. Fix TT2 stylesheets for buttons to enable compatibility with global styles added [here](https://github.com/WordPress/WordPress/commit/dac4e3be1196ccaf22647984269982dfee791056).
2. Introduce the `wp-element-button` class in a few more areas: 
- The _Order again_ button in the order-received page (See https://d.pr/i/oVXIas) and;
- The pagination buttons in the **My Account > Orders** page (See https://d.pr/i/lYs1jg)

### Expected behavior

- Buttons should play well with global styles and have appropriate spacings based on context, 
- All core's buttons should include the `wp-element-button` class.

### Actual behavior

Currently, when viewing the product catalog in TT2 and using a non-default style (e.g., the Blue one), buttons are not working as expected:
screencapture-twentytwentytwo-test-product-beanie-2023-04-05-17_25_02

### Steps to reproduce

-

### WordPress Environment

-

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by xristos3490**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR introduces the following changes:
- Wraps TT2's main button styles under the existence of the `woocommerce-block-theme-has-button-styles` class in the body.
- Replaces the `word-wrap: break-word` rule with the `white-space: normal`, to resolve issues with buttons in a table. See example: https://d.pr/i/xrjT9A
- Adds default color values in all CSS variables for buttons.
- Fixes button colors to be aligned with the colors described in the _theme.json_ file.

Here's an example of the end result in a single product view using the global _Blue_ style:
screencapture-twentytwentytwo-test-product-beanie-2023-04-05-17_42_36

Closes #37578

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:
 
**Testing for global styles:**

1. Create a site using the TT2 theme; or an existing one.
2. Change the global style to Blue and notice the button issues.
3. Apply this fix, and ensure that the buttons appear correctly.

### CODE REVIEW

File: plugins/woocommerce/client/legacy/css/twenty-twenty-two.scss [hunk 1]
```diff
@@ -53,6 +53,37 @@ $tt2-gray: #f7f7f7;
...
+	 * Buttons
+	 */
+	a.button,
+	button[name="add-to-cart"],
+	input[name="submit"],
+	button.single_add_to_cart_button,
+	button[type="submit"]:not(.wp-block-search__button) {
+		display: inline-block;
+		text-align: center;
+		white-space: normal;
```
Comment (Aljullu) [inline plugins/woocommerce/client/legacy/css/twenty-twenty-two.scss]: I guess you could skip `white-space: normal;` as it will be inherited from L150, no?

**Review by Aljullu — COMMENTED** [review]
Looking great! Thanks for working on this @xristos3490!

Some comments of things that I could find:

1. In the Shop page for some reason the space between the product title and the Add to Cart button is smaller now:

Before | After
--- | ---
imatge | imatge

2. Heads-up that the Mini Cart, Cart and Checkout blocks have the 'default' color styling:

imatge

3. In some other styles of the theme, the buttons changed their color completely. Is that something we want? Or should be considered a breaking change?

Before | After
--- | ---
imatge | imatge

**Comment by xristos3490** [comment]
Thanks for the review, @Aljullu! 🎉 

> In the Shop page for some reason the space between the product title and the Add to Cart button is smaller now:

Good catch! We probably need to keep "display" and "positioning" rules in place. I moved the `margin-top: 1em` back to the origin place in [this](https://github.com/woocommerce/woocommerce/pull/37579/commits/36b6b9ebaf4fd547f66df9744ee016728ebd0569) commit.

> Heads-up that the Mini Cart, Cart and Checkout blocks have the 'default' color styling:

What's the "default" state? Are you perhaps referring to the setting under **Styles > Colors > Buttons** see https://d.pr/i/4upL9e? Or is it the one under **Blocks > Button > Colors** see https://d.pr/i/ivRhMV

> In some other styles of the theme, the buttons changed their color completely. Is that something we want?

That's a great question! 

Tbh, up until now, TT2 included CSS variables and hardcoded colors for buttons in a way that didn't match the `theme.json` file. This obviously created a version of UI/UX that was stable but wrong. 

In conjunction with the previous discussion about "default colors", there are some [remaining aggressive styles](https://github.com/woocommerce/woocommerce/blob/7.5.1/plugins/woocommerce/client/legacy/css/twenty-twenty-two.scss#L136-L205) that are:
- Messing with the global styles; in a way that custom colors won't work since hardcoded rules reflect on the JSON file directly
- Looks like there were added to resolve the missing `wp-element-button` class in certain areas, like the ones this PR is fixing.

> Or should be considered a breaking change?

Yes, for sure. 

And since it's a breaking change, I would also remove all these aggressive rules. Removing them should enable global styles to work with custom colors out of the box.

I mean, if we are going to "break" it, let's at least do it _all the way_. Thoughts?

File: plugins/woocommerce/client/legacy/css/twenty-twenty-two.scss (hunk 1 above)
Comment (xristos3490) [reply plugins/woocommerce/client/legacy/css/twenty-twenty-two.scss]: Done in https://github.com/woocommerce/woocommerce/pull/37579/commits/2c3099e630c270990cbfac88ae3d052a88172418

**Comment by Aljullu** [comment]
> Good catch! We probably need to keep "display" and "positioning" rules in place. I moved the `margin-top: 1em` back to the origin place in [this](https://github.com/woocommerce/woocommerce/pull/37579/commits/36b6b9ebaf4fd547f66df9744ee016728ebd0569) commit.

Thanks!

> What's the "default" state? Are you perhaps referring to the setting under Styles > Colors > Buttons?

This. I mean, the color of any button with `wp-element-button` class. You can reproduce it if you create a post or page with an HTML block with these contents: `<button class="wp-element-button">bar</button>`. To me, it's a bit strange that the user can customize the colors of the buttons in `Styles > Colors > Buttons` but those settings are not applied in WC buttons because we override it. I know this wasn't introduced in this PR, just thinking out loud. This seems to be caused by [L205-218](https://github.com/woocommerce/woocommerce/pull/37579/files#diff-25bcda4a934f188daa9d7ff8746a2c877f0e2d3dd853ca53cd230f2cfae6ece9R205-R218).

Site Editor | Cart block | Shop page
--- | --- | ---
imatge | imatge | imatge


> I mean, if we are going to "break" it, let's at least do it _all the way_. Thoughts?

We will probably need somebody from Proton to make the last call on this. In general, I think we should try to avoid as much as possible changing the frontend of existing stores. But considering the current state was 'broken' with some style variations, it might make sense to change it. I'm not 100% sure about this.

**Comment by xristos3490** [comment]
As this seems tricky,[ I've extracted the missing button classes](https://github.com/woocommerce/woocommerce/pull/37579/commits/c32d6cd23a18661d39bdda4826bc8e2821e1c11e) into a separate PR to get things forward. You can find them here: https://github.com/woocommerce/woocommerce/pull/37933

**Comment by xristos3490** [comment]
> But considering the current state was 'broken' with some style variations, it might make sense to change it. I'm not 100% sure about this. This seems to be caused by [L205-218](https://github.com/woocommerce/woocommerce/pull/37579/files#diff-25bcda4a934f188daa9d7ff8746a2c877f0e2d3dd853ca53cd230f2cfae6ece9R205-R218).

I agree that the proper way to deal with global styles is to remove every button-related stylesheet and ensure that all buttons will include the required CSS classes; just like the work done [here](https://github.com/woocommerce/woocommerce/pull/37933).

Not sure either if we need to move forward with this. I'll wait for Proton's input :)

**Comment by xristos3490** [comment]
I will close this workstream due to the shift of relevant themes to TT3 and the risk to reward ratio.

**Pull request closed without merging**